metadata/proto/metadata.proto

```diff
@@ -90,6 +99,11 @@ service Api {
     rpc ListProviders(Empty) returns (stream Provider);
     rpc ListEntities(Empty) returns (stream Entity);
     rpc ListModels(Empty) returns (stream Model);
+    rpc ListTriggers(Empty) returns (stream Trigger);
+
+    rpc AddTrigger(TriggerRequest) returns (Empty);
```

Comment:
It's probably fine for now but generally, with rpcs, you want different requests for each.

So the better approach is to have an AddTriggerRequest and a RemoveTriggerRequest. Both of those can you have a TriggerRequest in them but the reason why you'd split it up is that programmers suck at determining what we need in the future so if you need to change the request for add now you're ALSO changing the request for remove; and with how annoying backwards compatability is within grpc it's best to safeguard against it.